scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ServerHandshaker.java

```diff
@@ -547,18 +549,23 @@ private void createServerHello(final ClientHello clientHello, final DTLSFlight f
 			serverHelloExtensions.addExtension(maxFragmentLengthExt);
 			LOGGER.debug(
 					"Negotiated max. fragment length [{} bytes] with peer [{}]",
-					new Object[]{maxFragmentLengthExt.getFragmentLength().length(), clientHello.getPeer()});
+					maxFragmentLengthExt.getFragmentLength().length(), clientHello.getPeer());
 		}
 
 		ServerNameExtension serverNameExt = clientHello.getServerNameExtension();
 		if (serverNameExt != null) {
-			// store the names indicated by peer for later reference during key exchange
-			indicatedServerNames = serverNameExt.getServerNames();
-			serverHelloExtensions.addExtension(ServerNameExtension.emptyServerNameIndication());
-			session.setSniSupported(true);
-			LOGGER.debug(
-					"Using server name indication received from peer [{}]",
-					clientHello.getPeer());
+			if (sniEnabled) {
+				// store the names indicated by peer for later reference during key exchange
+				indicatedServerNames = serverNameExt.getServerNames();
+				serverHelloExtensions.addExtension(ServerNameExtension.emptyServerNameIndication());
```

Comment:
Maybe we can add a comment here, that according RFC6066, section 3, page 7, the server SHALL include also the sni extension and the value SHALL be empty. Otherwise the comment above "store ... later" may be misinterpreted as conflict with the "emptyServerNameIndication".